homeassistant/components/totalconnect/config_flow.py

```diff
@@ -0,0 +1,59 @@
+"""Config flow for the Total Connect component."""
+import logging
+
+from total_connect_client import TotalConnectClient
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@config_entries.HANDLERS.register(DOMAIN)
+class TotalConnectConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Total Connect config flow."""
+
+    VERSION = 1
+
+    async def async_step_user(self, user_input=None):
+        """Handle a flow initiated by the user."""
+        errors = {}
+
+        if user_input is not None:
+            # Validate user input
+            username = user_input.get(CONF_USERNAME)
+            password = user_input.get(CONF_PASSWORD)
+
+            await self.async_set_unique_id(username)
+            self._abort_if_unique_id_configured()
+
+            valid = await self.is_valid(username, password)
+
+            if valid:
+                # authentication success / valid
+                return self.async_create_entry(
+                    title="Total Connect",
+                    data={CONF_USERNAME: username, CONF_PASSWORD: password},
+                )
+            # authentication failed / invalid
+            errors["base"] = "login"
+
+        data_schema = vol.Schema(
+            {vol.Required(CONF_USERNAME): str, vol.Required(CONF_PASSWORD): str}
+        )
+
+        return self.async_show_form(
+            step_id="user", data_schema=data_schema, errors=errors
+        )
+
+    async def async_step_import(self, user_input):
+        """Import a config entry."""
+        return await self.async_step_user(user_input)
+
+    async def is_valid(self, username="", password=""):
+        """Return true if the given username and password are valid."""
+        client = TotalConnectClient.TotalConnectClient(username, password)
```

Comment:
I/O in async context not allowed

Reply:
@Kane610 please help.  I don't understand what needs to be done here.  I need to do I/O to ensure the username and password are valid.  This follows the Hue component which is given as an example.  Do I need to put this I/O inside a hass.async_add_executor_job() call?

Reply:
Exactly, check out axis component which also has a sync library

Reply:
Thanks.  Think this is fixed now.